pkg/crc/machine/machine.go

```diff
@@ -73,6 +77,34 @@ func getCrcBundleInfo(bundlePath string) (*bundle.CrcBundleInfo, error) {
 	return bundle.Extract(bundlePath)
 }
 
+func checkDiskImageSha256sum(bundleinfo *bundle.CrcBundleInfo) error {
+	diskImagePath := bundleinfo.GetDiskImagePath()
+	expectedSha := bundleinfo.GetSha256sumOfDisk()
+	logging.Infof("Checking sha256sum of disk image %s ...", diskImagePath)
+	gotSha, err := getSha256Sum(diskImagePath)
+	if err != nil {
+		return err
+	}
+	if expectedSha != gotSha {
+		return fmt.Errorf("Expected sha %s Got %s", expectedSha, gotSha)
+	}
+	return nil
+}
+
+func getSha256Sum(filename string) (string, error) {
```

Comment:
I want this function to public but didn't find at which package it should be part of or a complete new pacakge.

Reply:
this is generic functionality to obtain a digest of a file, right? so either a namespace `files` or `digest`.

or perhaps even `validation`... but for that you need to provide the the compare to verification string and return a boolean.